tests/portForwardingServer.spec.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import http from 'http';
+
+import { contextTest as it, expect } from './config/browserTest';
+import type { LaunchOptions, ConnectOptions } from '../index';
+
+it.describe('forwarding proxy', () => {
+  it.skip(({ mode }) => mode !== 'default');
+
+  let targetTestServer: http.Server;
```

Comment:
Why don't we use the regular server? We never need two servers, right?

Reply:
We re-write ports on the client side to ensure that the request went through the protocol layer. With a single server in my opinion it's hard to proof that this request came actually from the client and not from the server side.